modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/portlet/shared/search/PortletSharedSearchRequestImpl.java

```diff
@@ -253,33 +265,45 @@ private SearchSettingsContributor _getSearchSettingsContributor(
 		);
 	}
 
-	private Stream<SearchSettingsContributor>
-		_getSearchSettingsContributorsStream(
-			ThemeDisplay themeDisplay, RenderRequest renderRequest) {
+	private List<SearchSettingsContributor> _getSearchSettingsContributors(
+		ThemeDisplay themeDisplay, RenderRequest renderRequest) {
 
-		Stream<Portlet> portletsStream = _getPortletsStream(
+		List<Portlet> portlets = _getPortlets(
 			themeDisplay.getLayout(), themeDisplay.getCompanyId());
 
-		return portletsStream.map(
-			portlet -> _getSearchSettingsContributorOptional(
-				portlet, themeDisplay, renderRequest)
-		).filter(
-			Optional::isPresent
-		).map(
-			Optional::get
-		);
+		List<SearchSettingsContributor> searchSettingsContributors =
+			new ArrayList<>();
+
+		for (Portlet portlet : portlets) {
+			Optional<SearchSettingsContributor>
```

Comment:
@slnn please get rid of this kind of Optional usage too. If the optional is at API level with many usages, you can leave them alone for now, But for all private usages, remove them all. You can just append commit to this pull.
Thanks!

Reply:
OK, got it

Reply:
Resend in https://github.com/liferay-core-infra/liferay-portal/pull/580